src/libraries/System.Text.Json/gen/JsonSourceGenerator.cs

```diff
@@ -39,90 +38,25 @@ public void Initialize(GeneratorInitializationContext context)
         /// <param name="executionContext"></param>
         public void Execute(GeneratorExecutionContext executionContext)
         {
-            Compilation compilation = executionContext.Compilation;
-
-            const string JsonSerializableAttributeName = "System.Text.Json.Serialization.JsonSerializableAttribute";
-            INamedTypeSymbol jsonSerializableAttribute = compilation.GetTypeByMetadataName(JsonSerializableAttributeName);
-            if (jsonSerializableAttribute == null)
-            {
-                return;
-            }
-
-            JsonSerializableSyntaxReceiver receiver = (JsonSerializableSyntaxReceiver)executionContext.SyntaxReceiver;
-            MetadataLoadContextInternal metadataLoadContext = new(compilation);
+            Parser parser = new(executionContext.Compilation);
+            SyntaxReceiver receiver = (SyntaxReceiver)executionContext.SyntaxReceiver;
+            _rootTypes = parser.GetRootSerializableTypes(receiver.CompilationUnits);
```

Comment:
Do you need to check if `reciever` is null here?

Comparing with
https://github.com/dotnet/runtime/blob/da580beb984f40516a408b8270e08d93daa0364e/src/libraries/Microsoft.Extensions.Logging/gen/LoggerMessageGenerator.cs#L27-L33

Reply:
Per the source generators [cookbook](https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md), it appears we don't have to check if the `receiver` is `null`:

```cs
// the generator infrastructure will create a receiver and populate it
// we can retrieve the populated instance via the context
MySyntaxReceiver syntaxReceiver = (MySyntaxReceiver)context.SyntaxReceiver;

// get the recorded user class
ClassDeclarationSyntax userClass = syntaxReceiver.ClassToAugment;
```

cc @jaredpar, @chsienki - do we need `null` checks for syntax receivers?